### PR TITLE
Filter the ruptures in the scenario calculator

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1420615.
The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/333/

### CODE REVIEW

File: openquake/commonlib/export/hazard.py [hunk 1]
```diff
@@ -195,21 +195,23 @@ def export_gmf_xml(key, export_dir, fname, sitecol, rupture_tags, gmfs):
...
     :rupture_tags: a list of rupture tags
     :gmfs: a dictionary of ground motion fields keyed by IMT
     """
     dest = os.path.join(export_dir, fname)
-    with floatformat('%12.8E'), open(dest, 'w') as f:
-        for imt, gmf in gmfs.iteritems():
-            for site, gmvs in zip(sitecol, gmf):
-                row = [imt, site.location.x, site.location.y] + list(gmvs)
-                f.write(scientificformat(row) + '\n')
+    dic = collections.defaultdict(list)
```
Comment (ptormene) [inline openquake/commonlib/export/hazard.py]: Please add an example

File: openquake/commonlib/tests/calculators/scenario_test.py [hunk 2]
```diff
@@ -88,3 +88,11 @@ def test_case_8(self):
         f1, f2 = self.frequencies(case_8, 0.5, 1.0)
         self.assertAlmostEqual(f1, 0)
         self.assertAlmostEqual(f2, 0)
+
+    @attr('qa', 'hazard', 'scenario')
+    def test_case_9(self):
+        # all sites were filtered out
+        out = self.run_calc(case_1.__file__, 'job.ini',
+                            maximum_distance=0.1, exports='csv')
+        self.assertEqualFiles(
```
Comment (ptormene) [inline openquake/commonlib/tests/calculators/scenario_test.py]: Please check if an error is expected with all sites are filtered out

File: openquake/commonlib/export/hazard.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/export/hazard.py]: Added a comment about the format of the CSV file

File: openquake/commonlib/tests/calculators/scenario_test.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/tests/calculators/scenario_test.py]: The comment is wrong, I have changed it.

File: openquake/commonlib/calculators/scenario.py [hunk 3]
```diff
@@ -70,9 +71,17 @@ def pre_execute(self):
         n_gmfs = self.oqparam.number_of_ground_motion_fields
         rupture = readinput.get_rupture(self.oqparam)
 
+        # filter the sites
+        self.sites = filters.filter_sites_by_distance_to_rupture(
+            rupture, self.oqparam.maximum_distance, self.sitecol)
+        if self.sites is None:
+            raise RuntimeError(
+                'All sites where filtered out! '
```
Comment (ptormene) [inline openquake/commonlib/calculators/scenario.py]: Typo: where -> were

**Comment by ptormene** [comment]
Apart from the very minor comment, LGTM

File: openquake/commonlib/calculators/scenario.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/scenario.py]: Ops! Fixed

**Pull request merged**